### PR TITLE
fix: sdk for artifact

### PR DESCRIPTION
**Pull request by thewbuk**
Because

- sdk implementation for artifact MVP

### CODE REVIEW

File: packages/sdk/src/vdp/artifact/ArtifactClient.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  async updateKnowledgeBase(
+    props: UpdateKnowledgeBaseRequest,
+  ): Promise<KnowledgeBase> {
+    const { ownerId, kbId, payload } = props;
+    try {
+      const data = await this._client.patch<{ knowledge_base: KnowledgeBase }>(
+        `/namespaces/${ownerId}/knowledge-bases/${kbId}`,
+        { body: JSON.stringify(payload) },
+      );
+      return Promise.resolve(data.knowledge_base);
```
Comment (EiffelFly) [inline packages/sdk/src/vdp/artifact/ArtifactClient.ts:72]: @thewbuk I think backend don't have snake_case in response right?

Comment (thewbuk) [reply packages/sdk/src/vdp/artifact/ArtifactClient.ts:72]: All APIs will be redone to catalogs format anyway

File: packages/sdk/src/vdp/artifact/ArtifactClient.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      return Promise.reject(error);
+    }
+  }
+
+  async createKnowledgeBase(
+    props: CreateKnowledgeBaseRequest,
+  ): Promise<KnowledgeBase> {
+    const { ownerId, payload } = props;
+    try {
+      const data = await this._client.post<{ knowledge_base: KnowledgeBase }>(
```
Comment (EiffelFly) [inline packages/sdk/src/vdp/artifact/ArtifactClient.ts:53]: @thewbuk Does backend still use snake_case here?

And please put this type as CreateKnowledgeBaseResponse

File: packages/sdk/src/vdp/artifact/ArtifactClient.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    }
+  }
+
+  async processKnowledgeBaseFiles(
+    props: ProcessKnowledgeBaseFilesRequest,
+  ): Promise<File[]> {
+    try {
+      const data = await this._client.post<{ files: File[] }>(
+        `/knowledge-bases/files/processAsync`,
+        { body: JSON.stringify({ file_uids: props.fileUids }) },
```
Comment (EiffelFly) [inline packages/sdk/src/vdp/artifact/ArtifactClient.ts:222]: @thewbuk Does backend still use snake_case in the request payload?

File: packages/toolkit/src/view/knowledge/components/CreateKnowledgeBaseCard.tsx [hunk 4]
```diff
@@ -112,23 +113,23 @@ export const CreateKnowledgeBaseCard = ({
...
       queryFn: () => useGetAllChunks(
         accessToken || "",
         knowledgeBase.ownerName,
         knowledgeBase.kbId,
         file.fileUid
       ),
-      enabled: enabledQuery && isHovered && !!files && !!accessToken,
+      enabled: enabledQuery && isHovered && !!files.data && !!accessToken,
     })),
   });
```
Comment (EiffelFly) [inline packages/toolkit/src/view/knowledge/components/CreateKnowledgeBaseCard.tsx:134]: @thewbuk Please don't directly write query function in the component. Extract it as a standalone query function

File: packages/toolkit/src/view/knowledge/components/tabs/CatalogFilesTab.tsx [hunk 5]
```diff
@@ -108,35 +103,31 @@ export const CatalogFilesTab = ({
     setIsFileDetailsOpen(false);
   };
 
-
   const handleDelete = (fileUid: string) => {
```
Comment (EiffelFly) [inline packages/toolkit/src/view/knowledge/components/tabs/CatalogFilesTab.tsx:106]: @thewbuk Before delete it, we should first check whether the query state is success

if (!file.isSuccess){
  return
}

**Pull request closed without merging**